### PR TITLE
feat(DEV-11776): [Demo app] assorted fixes

### PR DESCRIPTION
**Pull request by chebum**
This is a PR for multiple fixes for our auth demo app.

### CODE REVIEW

**Comment by chebum** [comment]
Discussed with @nekedos lack of e2e tests for the sdk-react package and agreed to implement some after 15.08.2024. A test for DataGrid rendering and scrolling should be done.

File: examples/with-nextjs-and-clerk-auth/src/themes/monite.ts [hunk 1]
```diff
@@ -7,6 +7,8 @@ import type {
 import type { Theme } from '@mui/material/styles/createTheme';
 import type { TypographyOptions } from '@mui/material/styles/createTypography';
 import { deepmerge } from '@mui/utils';
+import '@mui/x-data-grid/themeAugmentation';
```
Comment (saladNights) [inline examples/with-nextjs-and-clerk-auth/src/themes/monite.ts:10]: optional: could be written more explicitly
`import type {} from '@mui/x-data-grid/themeAugmentation';`

File: examples/with-nextjs-and-clerk-auth/src/themes/monite.ts [hunk 2]
```diff
@@ -7,6 +7,8 @@ import type {
...
+import { DataGridComponentNameToClassKey } from '@mui/x-data-grid/themeAugmentation/overrides';
```
Comment (saladNights) [inline examples/with-nextjs-and-clerk-auth/src/themes/monite.ts:11]: `DataGridComponentNameToClassKey` could be removed.
`themeAugmentation` is handling it.
I tested commands `build`, `typecheck`. Everything works without `DataGridComponentNameToClassKey`

File: examples/with-nextjs-and-clerk-auth/src/themes/monite.ts [hunk 3]
```diff
@@ -90,7 +93,9 @@ const typographyDark = deepmerge(typography, {
   },
 });
 
-const components: Components<Omit<Theme, 'components'>> = {
+const components:
+  | Components<Omit<Theme, 'components'>>
+  | DataGridComponentNameToClassKey = {
```
Comment (saladNights) [inline examples/with-nextjs-and-clerk-auth/src/themes/monite.ts:98]: `DataGridComponentNameToClassKey` could be removed

File: packages/sdk-react/src/ui/icons.tsx [hunk 4]
```diff
@@ -0,0 +1,19 @@
+/* eslint-disable lingui/no-unlocalized-strings */
+import { createSvgIcon } from '@mui/material';
```
Comment (saladNights) [inline packages/sdk-react/src/ui/icons.tsx:2]: Is this delete icon necessary? There is one in the current icon set.
If we want to introduce a new icon set and allow to custom it, let's find a more comprehensive approach

**Review by saladNights — CHANGES_REQUESTED** [review]

File: examples/with-nextjs-and-clerk-auth/src/components/OrganizationSwitcher.tsx [hunk 5]
```diff
@@ -5,26 +5,15 @@ import { ComponentProps } from 'react';
 import { normalizePathTrailingSlash } from 'next/dist/client/normalize-trailing-slash';
 import { usePathname } from 'next/navigation';
 
-import {
```
[the remark this answers is not in the document]
Comment (chebum) [reply examples/with-nextjs-and-clerk-auth/src/components/OrganizationSwitcher.tsx:8]: Yes, design projects require a switcher in this place to be visible.

File: examples/with-nextjs-and-clerk-auth/src/themes/monite.ts (hunk 1 above)
Comment (chebum) [reply examples/with-nextjs-and-clerk-auth/src/themes/monite.ts:10]: When import is written this way, the Intellij will optimize it away

**Review by vkuprin — APPROVED** [review]

**Pull request merged**